cmd/gen-docs/main.go

```diff
@@ -0,0 +1,134 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/AlecAivazis/survey/v2"
+	version "github.com/OctopusDeploy/cli"
+	"github.com/OctopusDeploy/cli/pkg/apiclient"
+	"github.com/OctopusDeploy/cli/pkg/cmd/root"
+	"github.com/OctopusDeploy/cli/pkg/factory"
+	"github.com/OctopusDeploy/cli/pkg/question"
+	"github.com/briandowns/spinner"
+	"github.com/spf13/cobra"
+	"github.com/spf13/cobra/doc"
+	"github.com/spf13/pflag"
+)
+
+type TemplateInformation struct {
+	Title   string
+	Command *cobra.Command
+}
+
+func main() {
+	if err := run(os.Args); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+func run(args []string) error {
+	flags := pflag.NewFlagSet("", pflag.ContinueOnError)
+	manPage := flags.BoolP("man-page", "", false, "Generate manual pages")
+	website := flags.BoolP("website", "", false, "Generate website pages")
+	dir := flags.StringP("doc-path", "", "", "Path directory where you want generate doc files")
+	help := flags.BoolP("help", "h", false, "Help about any command")
+
+	if err := flags.Parse(args); err != nil {
+		return err
+	}
+
+	if *help {
+		fmt.Fprintf(os.Stderr, "Usage of %s:\n\n%s", filepath.Base(args[0]), flags.FlagUsages())
+		return nil
+	}
+
+	if *dir == "" {
+		return fmt.Errorf("error: --doc-path not set")
+	}
+
+	askProvider := question.NewAskProvider(survey.AskOne)
+	clientFactory := apiclient.NewStubClientFactory()
+	s := spinner.New(spinner.CharSets[11], 100*time.Millisecond, spinner.WithColor("cyan"))
+	buildVersion := strings.TrimSpace(version.Version)
+	f := factory.New(clientFactory, askProvider, s, buildVersion)
+
+	cmd := root.NewCmdRoot(f, clientFactory, askProvider)
+	cmd.DisableAutoGenTag = true
+	cmd.InitDefaultHelpCmd()
+
+	if err := os.MkdirAll(*dir, 0755); err != nil {
```

Comment:
is `0755` safe here?
(I'm not criticizing; I genuinely don't know -- this seems like a bit of a hard question. If we're doing something like copying github then a comment explaining that would be helpful.)

Reply:
Adjusted to `0644` 😉